### PR TITLE
CLI-V2 pre prod #6

### PR DESCRIPTION
**Pull request by Mark-Labuschagne**
Don't review yet, these PRs are first being structured in the most convenient fashion.

### What this PR does
- introduces docker utility functions
- function to aggregate launch preparation functions
- exit handler

### Testing
- run ./unit-test.sh and check unit tests

### CODE REVIEW

File: cli/src/core/parse/validate.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	}
+
+	profilePackagesMap := make(map[string]bool)
+	for _, prof := range config.Profiles {
+		if prof.Name == profile {
+			for _, p := range prof.Packages {
+				profilePackagesMap[p] = true
+			}
+		}
+	}
```
Comment (bradsawadye) [inline cli/src/core/parse/validate.go:49]: Should we not have a break here?

Comment (Mark-Labuschagne) [reply cli/src/core/parse/validate.go]: Done in [c1c77e1](https://github.com/openhie/package-starter-kit/pull/36/commits/c1c77e1d241de4a827525696efd5af5d945fb93e)

File: cli/unit-test.sh [hunk 2]
```diff
@@ -7,3 +7,6 @@ FILE_PATH=$(
 
 cd "$FILE_PATH"/src/core/parse || exit
 go test .
+
+cd "$FILE_PATH"/src/util/slice || exit
+go test .
```
Comment (nour-borgi) [inline cli/unit-test.sh:12]: what about making these verbose ? 
```suggestion
go test . -v

cd "$FILE_PATH"/src/util/slice || exit
go test . -v
```

File: cli/src/util/slice/slice_test.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+func Test_SliceContains(t *testing.T) {
+	type testCase struct {
+		slice      interface{}
+		typeOf     string
+		matchValue interface{}
+		wantMatch  bool
+	}
+
+	testCases := []testCase{
```
Comment (nour-borgi) [inline cli/src/util/slice/slice_test.go:17]: missing a test case for empty slice ?

File: cli/src/core/parse/validate.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	if profileName != "" {
+		return validateProfile(cmd, profileName, config)
+	}
+
+	return nil
+}
+
+func validateProfile(cmd *cobra.Command, profileName string, config *core.Config) error {
+	profilePackagesMap := make(map[string]bool)
+	var profile core.Profile
```
Comment (nour-borgi) [inline cli/src/core/parse/validate.go:47]: I think profile is declared and not used

File: cli/src/core/parse/validate.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+				profilePackagesMap[p] = true
+				profile = prof
+			}
+			
+			break
+		}
+	}
+
+	if len(profilePackagesMap) < 1 {
+		return errors.Wrap(ErrNoSuchProfile, profileName)
```
Comment (nour-borgi) [inline cli/src/core/parse/validate.go:60]: I think this is a misleading error, the profile can exist with no packages in it
so maybe we can split two cases : 
- No such profile means the profile name does not exist in config file
- No packages defined in profile name 

Or we can edit the error message to be: no such profile or the profiles defined has no packages specified

File: cli/src/core/parse/validate.go [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	}
+
+	customPackages, err := cmd.Flags().GetStringSlice("custom-path")
+	if err != nil {
+		return errors.Wrap(err, "")
+	}
+
+	for _, cp := range customPackages {
+		packName := strings.TrimSuffix(path.Base(path.Clean(cp)), path.Ext(cp))
+		if profilePackagesMap[packName] {
```
Comment (nour-borgi) [inline cli/src/core/parse/validate.go:81]: I think here you're assuming that the package id is the the basename of the path
Which is not always right ?

File: cli/unit-test.sh (hunk 2 above)
Comment (Mark-Labuschagne) [reply cli/unit-test.sh:12]: I don't like doing verbose mode with running tests. When a test fails, it will still give output (without verbose mode). There is no reason to see output from passing tests, we only ever want to see output from failed tests.

File: cli/src/core/parse/validate.go (hunk 6 above)
Comment (Mark-Labuschagne) [reply cli/src/core/parse/validate.go:81]: If you look at all of the packages in platform, the `package-metadata.json` file will always have the same ID as it's containing folder, I think that is a good way for it to be structured. I will look through the test cases and make sure that this is the case, and that the CLI confirms to this always.

File: cli/src/core/parse/validate.go (hunk 4 above)
Comment (Mark-Labuschagne) [reply cli/src/core/parse/validate.go:47]: It is being used, check line 63

File: cli/src/util/slice/slice_test.go (hunk 3 above)
Comment (Mark-Labuschagne) [reply cli/src/util/slice/slice_test.go:17]: Done in [1874046](https://github.com/openhie/package-starter-kit/pull/36/commits/187404627b4d58a4b2259efa6385229c5a502e28)

File: cli/src/core/parse/validate.go (hunk 4 above)
Comment (Mark-Labuschagne) [reply cli/src/core/parse/validate.go:108]: Check again

File: cli/src/core/parse/validate.go (hunk 5 above)
Comment (Mark-Labuschagne) [reply cli/src/core/parse/validate.go:121]: Yes, you are entirely right! Fixed this in [f3a068c](https://github.com/openhie/package-starter-kit/pull/36/commits/f3a068cd8e75a7c5f8ebe34b5e58ad8ae8e525c6)

File: cli/src/core/parse/validate.go (hunk 6 above)
Comment (nour-borgi) [reply cli/src/core/parse/validate.go]: cool
